From 14b4787acd36f399c383f45917bc05c49a3926d7 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Mon, 4 Feb 2019 09:58:16 -0600
Subject: [PATCH 3/8] livedisplay: Minor code cleanup

* legacymm only:
 - Don't set display mode in constructor, this is handled by
   the backend when we don't have any non-backend handled modes
* legacymm and sdm:
 - Move delete[] tmp[i].name outside of the loop, since it needs
   to be deleted whether or not we get anything from the backend

Change-Id: I3ecac94a0affdbf9b587074879aa9a00ca8587c9
---
 legacymm/DisplayModes.cpp | 16 ++++------------
 sdm/DisplayModes.cpp      |  9 ++++-----
 2 files changed, 8 insertions(+), 17 deletions(-)

diff --git a/legacymm/DisplayModes.cpp b/legacymm/DisplayModes.cpp
index bebf2fe..bc9e2cb 100644
--- a/legacymm/DisplayModes.cpp
+++ b/legacymm/DisplayModes.cpp
@@ -43,13 +43,6 @@ DisplayModes::DisplayModes(void* libHandle) {
         dlsym(mLibHandle, "disp_api_get_default_display_mode"));
     disp_api_set_default_display_mode = reinterpret_cast<int (*)(int32_t, int)>(
         dlsym(mLibHandle, "disp_api_set_default_display_mode"));
-
-    if (isSupported()) {
-        DisplayMode mode = getDefaultDisplayModeInternal();
-        if (mode.id > 0) {
-            setDisplayMode(mode.id, false);
-        }
-    }
 }
 
 bool DisplayModes::isSupported() {
@@ -88,14 +81,13 @@ std::vector<DisplayMode> DisplayModes::getDisplayModesInternal() {
         if (disp_api_get_display_modes(0, 0, tmp, count) == 0) {
             for (int i = 0; i < count; i++) {
                 modes.push_back(DisplayMode{tmp[i].id, std::string(tmp[i].name)});
-                delete[] tmp[i].name;
-            }
-        } else {
-            for (int i = 0; i < count; i++) {
-                delete[] tmp[i].name;
             }
         }
 
+        for (int i = 0; i < count; i++) {
+            delete[] tmp[i].name;
+        }
+
         delete[] tmp;
     }
 
diff --git a/sdm/DisplayModes.cpp b/sdm/DisplayModes.cpp
index 4d68a8c..ac431f8 100644
--- a/sdm/DisplayModes.cpp
+++ b/sdm/DisplayModes.cpp
@@ -115,14 +115,13 @@ std::vector<DisplayMode> DisplayModes::getDisplayModesInternal() {
         if (disp_api_get_display_modes(mCookie, 0, 0, tmp, count, &flags) == 0) {
             for (int i = 0; i < count; i++) {
                 modes.push_back(DisplayMode{tmp[i].id, std::string(tmp[i].name)});
-                delete[] tmp[i].name;
-            }
-        } else {
-            for (int i = 0; i < count; i++) {
-                delete[] tmp[i].name;
             }
         }
 
+        for (int i = 0; i < count; i++) {
+            delete[] tmp[i].name;
+        }
+
         delete[] tmp;
     }
 
-- 
2.17.1

